native/src/main/java/io/ballerina/stdlib/http/api/HttpService.java

```diff
@@ -432,25 +437,62 @@ public static void populateInterceptorServicesRegistries(List<HTTPInterceptorSer
 
+        ServiceType balServiceType = (ServiceType) service.getBalService().getOriginalType();
+        boolean includesInterceptableService = balServiceType.getTypeIdSet().getIds().stream()
+                .anyMatch(typeId -> typeId.getName().equals(INTERCEPTABLE_SERVICE));
+        BArray interceptorsArrayFromService;
         BMap serviceConfig = getHttpServiceConfigAnnotation(service.getBalService());
-        if (Objects.isNull(serviceConfig)) {
-            service.setInterceptorServicesRegistries(interceptorServicesRegistries);
-            service.setBalInterceptorServicesArray(interceptorsArrayFromListener);
-            return;
-        }
+        if (includesInterceptableService) {
+            final Object[] createdInterceptors = new Object[1];
+            CountDownLatch latch = new CountDownLatch(1);
+            runtime.invokeMethodAsyncConcurrently(service.getBalService(), CREATE_INTERCEPTORS_FUNCTION_NAME, null,
+                    null, new Callback() {
+                @Override
+                public void notifySuccess(Object response) {
+                    if (response instanceof BError) {
+                        log.error("Error occurred while creating interceptors", response);
+                    } else {
+                        createdInterceptors[0] = response;
+                    }
+                    latch.countDown();
+                }
 
-        Object interceptorPipeline = serviceConfig.get(HttpConstants.ANN_INTERCEPTORS);
-        BArray interceptorsArrayFromService;
-        if (Objects.isNull(interceptorPipeline)) {
+                @Override
+                public void notifyFailure(BError bError) {
+                    bError.printStackTrace();
+                    System.exit(1);
+                }
+            }, null, PredefinedTypes.TYPE_ANY);
+            try {
+                latch.await();
+            } catch (InterruptedException exception) {
+                log.warn("Interrupted before getting the return type");
+            }
+
+            if (Objects.isNull(createdInterceptors[0]) || (createdInterceptors[0] instanceof BArray &&
+                    ((BArray) createdInterceptors[0]).size() == 0)) {
+                service.setInterceptorServicesRegistries(interceptorServicesRegistries);
+                service.setBalInterceptorServicesArray(interceptorsArrayFromListener);
+                return;
+            } else if (createdInterceptors[0] instanceof BArray) {
+                interceptorsArrayFromService = (BArray) createdInterceptors[0];
+            } else {
+                interceptorsArrayFromService = ValueCreator.createArrayValue(createdInterceptors,
+                        TypeCreator.createArrayType(((BObject) createdInterceptors[0]).getOriginalType()));
+            }
+        } else if (serviceConfig != null && serviceConfig.get(HttpConstants.ANN_INTERCEPTORS) != null) {
+            Object interceptorPipeline = serviceConfig.get(HttpConstants.ANN_INTERCEPTORS);
```

Comment:
Can we put a `TODO` here to remove this validation after removing the `interceptors` from the `ServiceConfig`?

Reply:
Updated in https://github.com/ballerina-platform/module-ballerina-http/pull/1606/commits/c7c84f058beac1f6d7a7e225d1ed6bd9cd57d182